lang: add cmakescript language

vicentebolea opened the pull request:
Fixes #5502 

In this PR I make an early attempt to add the CMakeScript as a language in compiler-explorer. I will be fantastic having this since CMake is one of the most used build system. Having an cmake interpreter in the web would probe truly useful for teams and individuals, it will be a fantastic way to share snippets of cmakescript and show how they are executed with different cmake versions.

vicentebolea commented:
@partouf

mattgodbolt commented on lib/compilers/cmakescript.ts:
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021, Compiler Explorer Authors
```
Can you set this to 2023 please

mattgodbolt commented on static/modes/asmcmakescript-mode.ts:
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021, Compiler Explorer Authors
```
Ditto

mattgodbolt reviewed (commented):
Nice! Some minor things, and ... can you add one example too please? That way when folks select CMakeScript they'll see at least one example.

vicentebolea commented:
Thanks @mattgodbolt for the quick review, I wonder if can you enable the CI to test this.

partouf commented on lib/compilers/_all.ts:
```diff
@@ -99,6 +99,7 @@ export {PythonCompiler} from './python.js';
 export {RacketCompiler} from './racket.js';
 export {RGACompiler} from './rga.js';
 export {RubyCompiler} from './ruby.js';
+export {CMakeScript} from './cmakescript.js';
```
Should be CMakeScriptCompiler

vicentebolea commented:
@partouf @mattgodbolt @DavidSpickett I have addressed all the comments here and it seems to run properly in my machine. Few questions

- It will be great to have a couple of versions deployed at godbolt so that the user can check how cmake behaves in different versions.
- `cmake -P` is a purely interpreted language meaning that we do not produce any type of compilation output. In that sense I have tweak the the CmakeCompiler to do the same thing in both compilation and execution phase (I made sure that it accepts users options in both cases).

Please let me know what you think, I am happy to keep improving this.

On another note from the CMake team they are happy with the naming of cmakescript as the language that we interpret with `cmake -P`

Thanks!

mattgodbolt commented on examples/cmakescript/default.cmake:
```diff
@@ -0,0 +1,13 @@
+# CMake can execute standalone CMake scripts with the -P parameter
```
Nice

mattgodbolt commented:
> It will be great to have a couple of versions deployed at godbolt so that the user can check how cmake behaves in different versions.

That's all under your control! Our infra repro controls the versions of CMake that are installed, and by suppling a `cmakescript.amazon.properties` file you can configure what versions are visible.

Though we might need to tweak a little thing in the infra installer (cc @partouf ) as currently we assume there's exactly one.

vicentebolea commented:
@mattgodbolt I see!

I wonder if you could approve the CI workflow.

mattgodbolt reviewed (commented):
Looks ok. I'll test locally and then we can get this merged. It won't show up on the site until we get a `cmakescript.amazon.properties` and solve the installation issues but, we're close!

mattgodbolt reviewed (approved):
Look good! I'll take a look at the installation thing

mattgodbolt commented:
@vicentebolea I'll configure some CMakes for the amazon setup :)

vicentebolea commented:
> @vicentebolea I'll configure some CMakes for the amazon setup :)

Thanks! We have an internal meeting in the cmake project this Wednesday it will be super awesome to show this to the team :zap:

mattgodbolt commented:
I'm trying to get this rolled out tonight :) I'll follow up here

mattgodbolt commented:
This is now live!
